packages/fields/src/types/Text/views/Field.js

```diff
@@ -1,21 +1,33 @@
 /** @jsx jsx */
 
 import { jsx } from '@emotion/core';
+import { useCallback, useMemo } from 'react';
 
 import { FieldContainer, FieldLabel, FieldDescription, FieldInput } from '@arch-ui/fields';
 import { Input } from '@arch-ui/input';
 
-const TextField = ({ onChange, autoFocus, field, errors, value = '', isDisabled }) => {
-  const handleChange = event => {
-    onChange(event.target.value);
-  };
+const isAccessDeniedError = error => error instanceof Error && error.name === 'AccessDeniedError';
```

Comment:
The changes here are mainly just moving things around / using `useCallback` and `useMemo`.